ouroboros-consensus-test/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -77,6 +92,217 @@ tests = testGroup "Mempool"
   , testProperty "removeTxs"                               prop_Mempool_removeTxs
   ]
 
+{-------------------------------------------------------------------------------
+  Mempool Pure Properties
+-------------------------------------------------------------------------------}
+
+-- | Test that removed transactions are not in the
+--   result transactionsIds of the returned internal state.
+prop_pure_removeTxs
+  :: TestSetupSub
+  -> Property
+prop_pure_removeTxs TestSetupSub {..} =
+    withInternalState tssSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (internalStRes, _)    = pureRemoveTxs (map txId tssSubTxs) mpArgs internalSt ledgerState
+          txIdsRemaining        = Set.toList (isTxIds internalStRes)
+      in property $ null $ intersect txIdsRemaining (map txId tssSubTxs)
+
+-- | Test that after removing transactions all the other transactions are still available.
+--
+prop_pure_removeTxs_contra
+    :: TestSetupWithTxs
+    -> Property
+prop_pure_removeTxs_contra TestSetupWithTxs {..} =
+    withInternalState testSetup $
+    \ mpArgs internalSt ledgerState ->
+      let (_res,internalSt')     = runTryAddTxs mpArgs internalSt (map fst txs)
+          (internalStRes, _tra)  = pureRemoveTxs (map (txId . fst) txs)
+                                      mpArgs internalSt' ledgerState
+          txIdsRemaining         = Set.toList (isTxIds internalStRes)
+      in sort txIdsRemaining === sort (Set.toList (isTxIds internalSt))
```

Comment:
```suggestion
      in sort txIdsRemaining === Set.toAscList (isTxIds internalSt)
```

Reply:
Actually, I would inline `txsIdsRemaining`, that makes it clearer what you're doing, IMO.
Alternatively, introduce a binding for the RHS (e.g., `txIdsRemaining'`) so that there's a nice symmetry between the two.